src/contract_wrappers/exchange_wrapper.ts

```diff
@@ -740,24 +807,40 @@ export class ExchangeWrapper extends ContractWrapper {
     }
     private async _isRoundingErrorAsync(takerTokenAmount: BigNumber.BigNumber,
                                         fillTakerAmount: BigNumber.BigNumber,
-                                        makerTokenAmount: BigNumber.BigNumber): Promise<boolean> {
+                                        makerTokenAmount: BigNumber.BigNumber,
+                                        exchangeContractAddress: string): Promise<boolean> {
         await assert.isUserAddressAvailableAsync(this._web3Wrapper);
-        const exchangeInstance = await this._getExchangeContractAsync();
+        const exchangeInstance = await this._getExchangeContractAsync(exchangeContractAddress);
         const isRoundingError = await exchangeInstance.isRoundingError.call(
             takerTokenAmount, fillTakerAmount, makerTokenAmount,
         );
         return isRoundingError;
     }
-    private async _getExchangeContractAsync(): Promise<ExchangeContract> {
-        if (!_.isUndefined(this._exchangeContractIfExists)) {
-            return this._exchangeContractIfExists;
+    private async _getExchangeContractAsync(exchangeContractAddress: string): Promise<ExchangeContract> {
+        if (!_.isUndefined(this._exchangeContractByAddress[exchangeContractAddress])) {
+            return this._exchangeContractByAddress[exchangeContractAddress];
         }
+        const ExchangeArtifacts = this._getExchangeArtifactsByAddressOrThrow(exchangeContractAddress);
         const contractInstance = await this._instantiateContractIfExistsAsync((ExchangeArtifacts as any));
-        this._exchangeContractIfExists = contractInstance as ExchangeContract;
-        return this._exchangeContractIfExists;
+        this._exchangeContractByAddress[exchangeContractAddress] = contractInstance as ExchangeContract;
+        return this._exchangeContractByAddress[exchangeContractAddress];
+    }
+    private _getExchangeArtifactsByAddressOrThrow(exchangeContractAddress: string): ContractArtifact {
+        const exchangeArtifacts = _.values<ContractArtifact>(ExchangeArtifactsByName);
+        for (const exchangeArtifact of exchangeArtifacts) {
+            const exchangeAddressesInArtifact = _.map(
+                _.values(exchangeArtifact.networks),
+                artifactsByNetwork => artifactsByNetwork.address,
```

Comment:
This variable name confuses me. `artifactsByNetwork ` sounds like a map of networkIds to artifacts. But then we are indexing into it with `.address`. Please re-name.